src/transformers/modeling_transfo_xl.py

```diff
@@ -991,21 +1040,28 @@ def forward(
             inputs_embeds=inputs_embeds,
             output_attentions=output_attentions,
             output_hidden_states=output_hidden_states,
+            return_tuple=return_tuple,
         )
 
         last_hidden = transformer_outputs[0]
         pred_hid = last_hidden[:, -tgt_len:]
         outputs = transformer_outputs[1:]
 
         softmax_output = self.crit(pred_hid, labels)
-        if labels is None:
-            softmax_output = softmax_output.view(bsz, tgt_len, -1)
-            outputs = [softmax_output] + outputs
-        else:
-            softmax_output = softmax_output.view(bsz, tgt_len - 1)
-            outputs = [softmax_output, None] + outputs
-
-        return outputs  # (loss), logits or None if labels is not None (speed up adaptive softmax), new_mems, (all hidden states), (all attentions)
+        prediction_scores = softmax_output.view(bsz, tgt_len, -1) if labels is None else ()
```

Comment:
Here, I can't set prediction_scores to None when labels are passed (which was the previous behavior) because it will be ignored in the ModelOutput (as all None things are). Set () for something different to avoid breaking changes.